### PR TITLE
feat: Add privacy on profile screen

### PR DESCRIPTION
**Pull request by jorelosorio**
Closes: https://github.com/AtB-AS/kundevendt/issues/8621

### CODE REVIEW

**Review by marius-at-atb — APPROVED** [review]
Lgtm! See some suggestions with nynorsk corrections and a couple of questions

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+          {isBeaconsSupported && (
+            <Button
+              style={style.spacing}
+              leftIcon={{svg: Delete}}
+              interactiveColor="interactive_destructive"
+              text={t(
+                PrivacySettingsTexts.sections.other.items.deleteData.title,
+              )}
+              onPress={() => {
+                deleteCollectedData();
```
[the remark this answers is not in the document]
Comment (jorelosorio) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx]: It was added.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+              icon="external-link"
+              accessibility={{
+                accessibilityHint: t(
+                  ProfileTexts.sections.privacy.linkSectionItems.privacy
+                    .a11yHint,
+                ),
+              }}
+              testID="privacyButton"
+              onPress={async () => {
+                kettleInfo?.privacyDashboardUrl &&
```
[the remark this answers is not in the document]
Comment (jorelosorio) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx]: Fixed

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx [hunk 3]
```diff
@@ -30,6 +30,10 @@ export const Profile_PrivacyScreen = () => {
   const {privacy_policy_url} = useRemoteConfig();
   const style = useStyle();
   const {clearHistory} = useSearchHistory();
+  const [isCleaningCollectedData, setIsCleaningCollectedData] =
+    React.useState<boolean>(false);
+  const [isCleaningHistorydData, setIsCleaningHistoryData] =
```
Comment (marius-at-atb) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx:35]: Typo: isCleaningHistorydData -> isCleaningHistoryData

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { translation as _ } from '../../commons';
+const PrivacySettingsTexts = {
+    sections: {
+        items: {
+            clarCollectedData: {
+                title: _('Tøm innsamlet data', 'Clear collected data', 'Tøm innsamlet data'),
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:6]:
```suggestion
                title: _('Tøm innsamlet data', 'Clear collected data', 'Tøm innsamla data'),
```

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            },
+            controlPanel: {
+                title: _('Kontrollpanel for datainnsamling', 'Control panel for data collection', 'Kontrollpanel for datainnsamling'),
+                subTitle: _(
+                    'Du kan se og laste ned data som blir samlet  om dine reisevaner. Det gjelder også data om busser og holdeplasser du er i nærheten av.',
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:11]:
```suggestion
                    'Du kan se og laste ned data som blir samlet om dine reisevaner. Det gjelder også data om busser og holdeplasser du er i nærheten av.',
```

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                    'You can view and download data that is collected about your travel habits. This also applies to data about buses and stops you are close to.',
+                    'Du kan se og laste ned data som blir samlet  om dine reisevaner. Det gjelder også data om busser og holdeplasser du er i nærheten av.'
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:13]:
```suggestion
                    'Du kan sjå og laste ned data som blir samla om reisevanane dine. Det gjeld også data om bussar og haldeplassar du er i nærleiken av.'
```

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                ),
+                a11yHint: _(
+                    'Aktivér for å gå til kontrollpanel for datainnsamling (ekstern side)',
+                    'Activate to control panel for data collection (external content)',
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:17]:
```suggestion
                    'Activate to go to control panel for data collection (external content)',
```

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+                    'Aktiver for å gå til kontrollpanel for datainnsamling (ekstern side)',
+                ),
+            }
+        },
+        consents: {
+            title: _('Samtykke', 'Consent', 'Samtykke'),
+            items: {
+                CollectTravelHabits: {
+                    title: _('Samle reisevaner', 'Collect travel habits', 'Samle reisevaner'),
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:26]:
```suggestion
                    title: _('Samle reisevaner', 'Collect travel habits', 'Samle reisevanar'),
```

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                    subText: _(
+                        'Du gir oss samtykke til å samle og analysere dine reisevaner ved hjelp av Bluetooth, din posisjon, bevegelses- og treningsaktivitet. Vi kan for eksempel samle data om holdeplasser i nærheten, busser du tar, hvor du reiser og om du går, sykler eller reiser med bil. Dataen brukes til å forbedre reisetilbudet ditt i Trøndelag, Samtykket lar oss sende deg undersøkelser som er relevante for deg.',
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:28]:
```suggestion
                        'Du gir oss samtykke til å samle og analysere dine reisevaner ved hjelp av Bluetooth, din posisjon, bevegelses- og treningsaktivitet. Vi kan for eksempel samle data om holdeplasser i nærheten, busser du tar, hvor du reiser og om du går, sykler eller reiser med bil. Dataene brukes til å forbedre reisetilbudet ditt i Trøndelag. Samtykket lar oss sende deg undersøkelser som er relevante for deg.',
```

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+                        'You give us consent to collect and analyze your travel habits using Bluetooth, your location, movement and exercise activity. For example, we may collect data about nearby stops, buses you take, where you travel and whether you walk, cycle or travel by car. The data is used to improve your travel offer in Trøndelag. The consent allows us to send you surveys that are relevant to you.',
+                        'Du gir oss samtykke til å samle og analysere dine reisevaner ved hjelp av Bluetooth, din posisjon, bevegelses- og treningsaktivitet. Vi kan for eksempel samle data om holdeplasser i nærheten, busser du tar, hvor du reiser og om du går, sykler eller reiser med bil. Dataen brukes til å forbedre reisetilbudet ditt i Trøndelag, Samtykket lar oss sende deg undersøkelser som er relevante for deg.',
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:30]:
```suggestion
                        'Du gir oss samtykke til å samle og analysere reisevanane dine ved hjelp av Bluetooth, din posisjon, bevegings- og treningsaktivitet. Vi kan for eksempel samle data om haldeplassar i nærleiken, bussar du tek, kvar du reiser og om du går, syklar eller reiser med bil. Dataene brukast til å forbetre reisetilbodet ditt i Trøndelag. Samtykket lar oss sende deg undersøkingar som er relevante for deg.',
```

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+                    ),
+                },
+            },
+        },
+    },
+    clearCollectedData: {
+        label: _(
+            'Tøm innsamlet data',
+            'Clear collected data',
+            'Tøm innsamlet data',
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:40]:
```suggestion
            'Tøm innsamla data',
```

File: src/translations/screens/subscreens/PrivacySettingsTexts.ts [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+        ),
+        a11yHint: _(
+            'Aktivér for å tøm innsamlet data',
+            'Activate to clear collected data',
+            'Klikk for å tøm innsamla data',
```
Comment (marius-at-atb) [inline src/translations/screens/subscreens/PrivacySettingsTexts.ts:45]:
```suggestion
            'Aktivér for å tømme innsamlet data',
            'Activate to clear collected data',
            'Aktivér for å tømme innsamla data',
```

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+    deleteCollectedData,
+    isBeaconsSupported,
+  } = useBeacons();
+  const {privacy_policy_url} = useRemoteConfig();
+  const style = useStyle();
+  const {clearHistory} = useSearchHistory();
+  const [isCleaningCollectedData, setIsCleaningCollectedData] =
+    React.useState<boolean>(false);
+  const [isCleaningHistoryData, setIsCleaningHistoryData] =
+    React.useState<boolean>(false);
```
Comment (gorandalum) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx:36]: Does these actions need a loading spinner? Especially the one with cleaning history (which is just local on the device) should be pretty immediate. The general notion is to add a loading spinner if anything takes a significant time for the user. 

A couple articles on the matter:

https://ux.stackexchange.com/questions/92092/what-is-the-minimum-waiting-time-after-when-should-a-page-loader-be-displayed-to
https://uxmovement.com/navigation/progress-bars-vs-spinners-when-to-use-which

**Review by gorandalum — APPROVED** [review]
Nice work :+1:

**Pull request merged**